Contributions/Using_Linux_in_a_Safe_System.md

```diff
@@ -0,0 +1,348 @@
+Solutions are often heavily contextualised, and must be designed to suit specific
+needs, in terms of use cases, safety requirements and system design.
+
+The evaluation of the very same mitigation strategy can range from overkill to
+insufficient, depending on where/how it is applied.
+
+
+## **Structure of the document**
+- The document starts with a section containing design considerations.
+- The following section contains several examples of how to apply such
+  considerations to the design of the Linux-bearing system.
+
+
+## **Considerations**
+
+**Premise:** some of the following considerations assume a fairly high level
+of freedom in designing the overall system.
+While one must acknowledge that in practice this freedom will not be always
+available, due to design constraints, it is equally true that attempting to
+cope with these constraints can lead to an even higher total cost of ownership.
+
+* **Clearly define requirements toward system integrity and availability, upfront**
+
+  Having to simply report failures (within a maximum allotted time - and how much?)
+  usually is significantly simpler than having to guarantee that a certain
+  functionality will not be interrupted.
+
+  Clarifying the scope of the design goes a long way in defining criteria for
+  identifying acceptable solutions.
+
+* **Linux is intrinsically not safe**
+
+  While Linux has many remarkable properties, safety is not one of them, which means
+  that it won't be possible to just allocate safety requirements to Linux, and expect
+  them to be automatically satisfied. Quite the opposite, actually.
+
+  Having to cope with safety requirements assigned to Linux will involve a price to
+  be paid, in one form of another.
+
+  The problem with said cost is that it might *also* affect negatively the goals
+  that led to the selection of Linux, in the first place. For example, if Linux was
+  selected because of its a rapid turnaround in fixing security bugs, selecting
+  an approach that would introduce an lengthy qualification process would undermine
+  the advantage of the rapid turnaround.
+
+* **Aggressively placing safety requirements on anything but (components of) the Linux kernel**
+
+  The natural consequence of the previous point is that one should explore - given a
+  specific set of safety requirements for the system as a whole, if they could be
+  satisfied through some component other than Linux (more on this later).
+
+  There are several additional reasons for favouring an external component, besides
+  what has been mentioned already:
+  *  much greater probability for the component to be simpler (and thus easier to qualify)
+  *  opportunity to have a better control over the component (e.g. use a software
+     architecture that is better suited for safety, formalised processes, etc.)
+  *  decouple release cycles of the Linux kernel from the safety component
+  *  other non-technical reasons (e.g. different licensing)
+
```

Comment:
Here I would add a sentence saying "indeed the drawback is that the more the components that are present in the system the higher the cost to be paid is"

Reply:
No, that argumentation is incomplete and misleading, one must consider the total cost of ownership: adding one single component and enabling safety claims on that one can be less costly than not adding components and having to make the existing ones safe.

Reply:
Gab, what about:

"However, adding additional components to a system can also increase the costs of production or maintenance, or make the associated safety arguments more difficult to follow, so it will be necessary to find the appropriate balance for any given system."